HermaFx.Foundation/RuntimeExtensions/GenericExtensions.cs

```diff
@@ -135,5 +135,22 @@ public static T ThrowIfNull<T>(this T source, string paramName, string message,
 
 			return source;
 		}
+
+		public static string ThrowIfNullOrWhiteSpace(this string @this, string message, params object[] args)
+		{
+			if (string.IsNullOrWhiteSpace(@this))
+				throw new ArgumentNullException(args != null ? string.Format(message, args) : message);
+
+			return @this;
+		}
+
+		public static T ThrowIfDefault<T>(this T @this, string message, params object[] args)
```

Comment:
this is more like a Guard, than an extension, so probably this should be placed at a new file for Guards (if there is not already an existing one).

Reply:
Mmm, there are already two similar methods here..

1. public static T ThrowIfNull<T>(this T source)
2. public static T ThrowIfNull<T>(this T source, string paramName)

That's why i chose this class.
Shall we move all of them?